Dockerfile

```diff
@@ -0,0 +1,134 @@
+COPY --chown=goblin:goblin packages/backend/native-utils/src/lib.rs packages/backend/native-utils/src/
+COPY --chown=goblin:goblin packages/backend/native-utils/migration/Cargo.toml packages/backend/native-utils/migration/Cargo.toml
+COPY --chown=goblin:goblin packages/backend/native-utils/migration/src/lib.rs packages/backend/native-utils/migration/src/
+
+# Install cargo dependencies
+RUN cargo fetch --locked --manifest-path /goblin/packages/backend/native-utils/Cargo.toml
+
+
+# Copy only the dependency-related files first, to cache efficiently
+COPY --chown=goblin:goblin package.json pnpm*.yaml ./
+COPY --chown=goblin:goblin packages/backend/package.json packages/backend/package.json
+COPY --chown=goblin:goblin packages/client/package.json packages/client/package.json
+COPY --chown=goblin:goblin packages/sw/package.json packages/sw/package.json
+COPY --chown=goblin:goblin packages/firefish-js/package.json packages/firefish-js/package.json
+COPY --chown=goblin:goblin packages/megalodon/package.json packages/megalodon/package.json
+COPY --chown=goblin:goblin packages/backend/native-utils/package.json packages/backend/native-utils/package.json
+COPY --chown=goblin:goblin packages/backend/native-utils/npm/linux-x64-musl/package.json packages/backend/native-utils/npm/linux-x64-musl/package.json
+COPY --chown=goblin:goblin packages/backend/native-utils/npm/linux-arm64-musl/package.json packages/backend/native-utils/npm/linux-arm64-musl/package.json
+
+# Configure corepack and pnpm, and install dev mode dependencies for compilation
+RUN corepack install -g pnpm@8.15.4
+# && corepack use pnpm@8.12.0
+RUN pnpm i --frozen-lockfile \
+    && pnpm prune
+
+# Copy in the rest of the native-utils rust files
+COPY --chown=goblin:goblin packages/backend/native-utils packages/backend/native-utils/
+
+# Compile native-utils
+RUN pnpm run --filter native-utils build
+
+# Copy in the rest of the files to compile
+COPY --chown=goblin:goblin . ./
+RUN env NODE_ENV=production sh -c "pnpm run --filter '!native-utils' build && pnpm run gulp"
+
+# Trim down the dependencies to only those for production
+RUN pnpm i --prod --frozen-lockfile \
+    && pnpm prune
+
+
+############################# Runtime container
+FROM debian:bookworm-20240211
+ENV PATH="${PATH}:/home/goblin/.npm/bin:/home/goblin/nodejs/bin"
+
+WORKDIR /goblin
+
+# Create a user and install runtime dependencies
+RUN groupadd --gid 1000 goblin \
+    && useradd --uid 1000 --gid goblin --shell /bin/bash --create-home goblin \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
+        ca-certificates \
+        curl \
+        less \
+        mc \
+        nano \
+        vim \
+        procps \
+        libc6 \
+    && apt-get autoremove -y \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+#RUN apt update && apt install -y --no-install-recommends \
+#    ffmpeg  \
+#    vips-dev  \ #TODO
```

Comment:
Ah that reminds me - what's our opinion on video? I haven't tested uploading any video, should probably check these dependencies, although it would probably bump the image size by another 200-500MB.

Reply:
I haven't tried to do anything with video yet. It might be worth an option, because it really does seem like a feature parity thing with, well, everything. But also I only paid for 50 gigs of storage.

Reply:
at this point, you can't upload video. I mean, you probably can somehow, but it won't be added to a post (it only allows adding youtube videos at the moment).
my idea is to allow hot-linking mp4 files from somewhere else, but I'm not sure about hosting video on its own: Any server allowing it could be out of space very very quickly. Also I'm kind of burned out from having to work with the tumblr video encoding server, which was a pain in the ass to deal with from a UI perspective, so ... yeah, this is a problem for a potential future where Goblin is a widely used platform and people start demanding self-hosted video, but not in the short term :)

Reply:
@sirilyan and yeah, it's a feature parity issue... but media hosting is one of the main issues that main tumblr so expensive to run,  so ... maybe that's an area where we can wait for that parity :D

Reply:
Got it, will add video support later once i figure out a smooth and reliable way to disable it as default.